AndrewTehan/hometask-4/my_array_rotate.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+# this class is extension for Array
+class Array
+  def lab_rotate(offset = 1)
+    shift = offset % size
+
+    return self if shift.zero?
+
+    first_chunk = self[0, shift]
+    second_chunk = self[shift, size - 1]
+    second_chunk.append(*first_chunk)
+  end
+
+  def lab_rotate!(offset = 1)
+    replace(lab_rotate(offset))
```

Comment:
I'd suggest the other way of its solution since `replace` is a quite time-consuming operation. But it's not required to change it there 👍🏻